core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSource.java

```diff
@@ -70,4 +72,9 @@ void close()
     {
         return NOT_BLOCKED;
     }
+
+    default Map<String, Metric> getCustomMetrics()
```

Comment:
Call this `getMetrics()`.

Also, it needs to document the semantics of the returned metrics -- i.e., they are a new snapshot of the metrics, metrics need to be immutable, etc.